lm_eval/tasks/ds1000.py

```diff
@@ -0,0 +1,186 @@
+"""
+DS-1000: A Natural and Reliable Benchmark for Data Science Code Generation
+
+https://arxiv.org/pdf/2211.11501.pdf
+
+DS-1000 is a code generation benchmark with a thousand data science questions spanning seven Python libraries that (1) reflects diverse, realistic, and practical use cases, (2) has a reliable metric, (3) defends against memorization by perturbing questions.
+
+Homepage: https://ds1000-code-gen.github.io/
+"""
+
+import io, itertools, functools, pathlib, requests, warnings, zipfile
+from lm_eval.base import Task
+
+_CITATION = """
+@article{Lai2022DS1000,
+  title={DS-1000: A Natural and Reliable Benchmark for Data Science Code Generation},
+  author={Yuhang Lai and Chengxi Li and Yiming Wang and Tianyi Zhang and Ruiqi Zhong and Luke Zettlemoyer and Scott Wen-tau Yih and Daniel Fried and Sida Wang and Tao Yu},
+  journal={ArXiv},
+  year={2022},
+  volume={abs/2211.11501}
+}
+"""
+
+
+def get_tasks():
+    def _get_task(key):
+        class DS1000(DS1000General):
+            def __init__(self):
+                super().__init__(key)
+
+        return DS1000
+
+    return {
+        f"ds1000-{key.lower()}": _get_task(key)
+        for key in [
+            "All",
+            "Numpy",
+            "Pandas",
+            "Scipy",
+            "Matplotlib",
+            "Sklearn",
+            "Tensorflow",
+            "Pytorch",
+        ]
+    }
+
+
+class DS1000General(Task):
+    DATASET_PATH = None
+    DATASET_NAME = None
+
+    def __init__(self, key):
+        super().__init__(
+            stop_words=["</code>", "END SOLUTION", "\n\n"],
+            requires_execution=True,
+        )
+        self._key = key
+        self._dir = pathlib.Path(__file__).parent / "ds"
+        self._dir.mkdir(parents=True, exist_ok=True)
+        self._src = self._dir / "ds1000.py"
+        self._data = self._dir / "ds1000_data"
+        self._download_source()
+        self._download_dataset()
+        self._install_dependencies()
+
+    def _download_source(self):
+        url = "https://github.com/HKUNLP/DS-1000/blob/main/ds1000.py?raw=true"
```

Comment:
Can we use the hash of a commit to always download the same version for both source code and data, like:
```python
url = "https://github.com/HKUNLP/DS-1000/blob/49c1c543ada8b58138181333cdc62e613204efcf/ds1000.py?raw=true"
```
we should probably also add a warning about the data and source code being downloaded

Reply:
Yes, good point, addressed [here](https://github.com/bigcode-project/bigcode-evaluation-harness/pull/39/commits/0767073de0bfd18bcd84999a8a901421b158e602).